pkg/agent/sysadvisor/plugin/overcommitmentaware/reporter/reporter.go

```diff
@@ -0,0 +1,318 @@
+	}
+
+	o.ctx, o.cancel = context.WithCancel(context.Background())
+	return
+}
+
+func (o *OvercommitRatioReporterPlugin) Stop() error {
+	o.Lock()
+	defer func() {
+		o.started = false
+		o.Unlock()
+	}()
+
+	if !o.started {
+		return nil
+	}
+
+	o.cancel()
+	return nil
+}
+
+// GetReportContent get overcommitment ratio from manager directly.
+// Since the metrics collected by Manager are already an average within a time period,
+// we expect a faster response to node load fluctuations to avoid excessive overcommit of online resources.
+func (o *OvercommitRatioReporterPlugin) GetReportContent(_ context.Context, _ *v1alpha1.Empty) (*v1alpha1.GetReportContentResponse, error) {
+	response := &v1alpha1.GetReportContentResponse{
+		Content: []*v1alpha1.ReportContent{},
+	}
+
+	overcommitRatioMap, err := o.manager.GetOvercommitRatio()
+	if err != nil {
+		klog.Errorf("OvercommitRatioReporterPlugin GetOvercommitMent fail: %v", err)
+		return nil, err
+	}
+	klog.V(6).Infof("reporter get overcommit ratio: %v", overcommitRatioMap)
+
+	// overcommit data to CNR data
+	overcommitRatioContent, err := o.overcommitRatioToCNRAnnotation(overcommitRatioMap)
+	if err != nil {
+		return nil, err
+	}
+	response.Content = append(response.Content, overcommitRatioContent)
+
+	// get topologyProvider and guaranteed cpus
+	topologyProviderContent, err := o.getTopologyProviderReportContent(o.ctx)
+	if err != nil {
+		return nil, err
+	}
+	response.Content = append(response.Content, topologyProviderContent)
+
+	return response, nil
+}
+
+func (o *OvercommitRatioReporterPlugin) ListAndWatchReportContent(_ *v1alpha1.Empty, server v1alpha1.ReporterPlugin_ListAndWatchReportContentServer) error {
+	for {
+		select {
+		case <-o.ctx.Done():
+			return nil
+		case <-server.Context().Done():
+			return nil
+		}
+	}
+}
+
+func (o *OvercommitRatioReporterPlugin) overcommitRatioToCNRAnnotation(overcommitRatioMap map[v1.ResourceName]float64) (*v1alpha1.ReportContent, error) {
+	if len(overcommitRatioMap) <= 0 {
+		return nil, nil
+	}
+
+	klog.V(6).Infof("overcommitRatioMap: %v", overcommitRatioMap)
+	overcommitRatioAnnotation := map[string]string{}
+	for resource := range overcommitRatioMap {
+		switch resource {
+		case v1.ResourceCPU, consts.NodeAnnotationCPUOvercommitRatioKey:
+			overcommitRatioAnnotation[consts.NodeAnnotationCPUOvercommitRatioKey] = fmt.Sprintf("%.2f", overcommitRatioMap[resource])
+		case v1.ResourceMemory, consts.NodeAnnotationMemoryOvercommitRatioKey:
+			overcommitRatioAnnotation[consts.NodeAnnotationMemoryOvercommitRatioKey] = fmt.Sprintf("%.2f", overcommitRatioMap[resource])
+		default:
+			klog.Warningf("unknown resource: %v", resource.String())
+		}
+	}
+
+	value, err := json.Marshal(&overcommitRatioAnnotation)
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1alpha1.ReportContent{
+		GroupVersionKind: &util.CNRGroupVersionKind,
+		Field: []*v1alpha1.ReportField{
+			{
+				FieldType: v1alpha1.FieldType_Metadata,
+				FieldName: "Annotations",
```

Comment:
this "Annotations" can use util.CNRFieldNameAnnotations also

Reply:
> this "Annotations" can use util.CNRFieldNameAnnotations also

Done.